rad_gen.py

```diff
@@ -1,3 +1,5 @@
+from __future__ import annotations
```

Comment:
Do you still need this line, since I think you set the Python version to a high enough version? I could be wrong though.